File: JavaVer/src/Encryptor.java
import java.io.File;
import java.io.FileDescriptor;
import java.security.PublicKey;
import java.util.Base64;

import javax.crypto.SecretKey;

public class Encryptor {
    public static void main(String[] args){
        File publicKeyFile = new File(Constants.PUBLIC_KEY);

        File input = new File(Constants.FILE_TO_ENCRYPT);
        File output = new File(Constants.ENCRYPTED_FILE);

        byte[] content = FileUtil.readBytesFromFile(input);
        try {
            // 1. Generate AES key
            SecretKey key = AESUtil.generateAESKey();

            // 2. Encrypt file content
            byte[] encryptedContent = AESUtil.encryptFile(key, content);

            // 3. Encrypt AES key
            PublicKey publicKey = RSAUtil.getPublicKey(Base64.getDecoder().decode(FileUtil.readBytesFromFile(publicKeyFile)));
            byte[] encryptedKey = RSAUtil.encryptKey(publicKey, key.getEncoded());

            // 4. Attach encrypted key to file
            byte[] fileOutputContent = FileUtil.combineBytes(encryptedKey, encryptedContent);

            // 5. Write to output
            FileUtil.writeToFile(output, fileOutputContent);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}

File: CSharpVer/CSharpVer/Encryptor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Threading.Tasks;

namespace CSharpVer
{
    class Encryptor
    {
        public static void Main()
        {
            byte[] content = FileUtil.ReadBytesFromFile(Constants.FILE_TO_ENCRYPT);

            // 1. Generate AES key
            AesManaged aes = new AesManaged();
            aes.KeySize = 256;
            aes.BlockSize = 128;
            aes.Mode = CipherMode.CBC;
            aes.Padding = PaddingMode.PKCS7;
            aes.GenerateKey();
            aes.GenerateIV();

            byte[] aesKey = aes.Key;
            byte[] ivParams = aes.IV;

            Console.WriteLine(Convert.ToBase64String(aesKey));
            Console.WriteLine(Convert.ToBase64String(ivParams));

            // 2. Encrypt file content
            byte[] encryptedContent = AESUtil.EncryptFile(aesKey, ivParams, content);
           
            // 3. Encrypt AES key
            byte[] publicKey = FileUtil.ReadBytesFromFile(Constants.PUBLIC_KEY);
            byte[] encryptedKey = RSAUtil.EncryptKey(publicKey, aesKey);

            // 4. Attach encrypted key to file
            byte[] fileOutputContent = FileUtil.combineBytes(encryptedKey, encryptedContent);

            // 5. Write to output
            FileUtil.WriteToFile(Constants.ENCRYPTED_FILE, fileOutputContent);

            Console.ReadLine();
        }
    }
}
